nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,324 @@
+        .defaultValue("UTF-8")
+        .build();
+
+    static final List<PropertyDescriptor> descriptors;
+
+    static {
+        List<PropertyDescriptor> _temp = new ArrayList<>();
+        _temp.add(CONNECTION_SERVICE);
+        _temp.add(URI);
+        _temp.add(DB_ACCESS_KEY);
+        _temp.add(CONSISTENCY);
+        _temp.add(DATABASE_NAME);
+        _temp.add(CONTAINER_ID);
+        _temp.add(PARTITION_KEY);
+        descriptors = Collections.unmodifiableList(_temp);
+    }
+
+    protected CosmosClient cosmosClient;
+    protected CosmosContainer container;
+    protected AzureCosmosDBConnectionService connectionService = null;
+
+    @OnScheduled
+    public void createClient(ProcessContext context) throws Exception {
+        final String uri, accessKey;
+
+        if (context.getProperty(CONNECTION_SERVICE).isSet()) {
+            connectionService = context.getProperty(CONNECTION_SERVICE).asControllerService(AzureCosmosDBConnectionService.class);
+            this.cosmosClient = this.connectionService.getCosmosClient();
+        } else {
+            uri =  context.getProperty(URI).getValue();
+            accessKey = context.getProperty(DB_ACCESS_KEY).getValue();
+            final ConsistencyLevel clevel;
+            final String selectedConsistency = context.getProperty(CONSISTENCY).getValue();
+
+            switch(selectedConsistency) {
+                case CONSISTENCY_STRONG:
+                    clevel =  ConsistencyLevel.STRONG;
+                    break;
+                case CONSISTENCY_CONSISTENT_PREFIX:
+                    clevel = ConsistencyLevel.CONSISTENT_PREFIX;
+                    break;
+                case CONSISTENCY_SESSION:
+                    clevel = ConsistencyLevel.SESSION;
+                    break;
+                case CONSISTENCY_BOUNDED_STALENESS:
+                    clevel = ConsistencyLevel.BOUNDED_STALENESS;
+                    break;
+                case CONSISTENCY_EVENTUAL:
+                    clevel = ConsistencyLevel.EVENTUAL;
+                    break;
+                default:
+                    clevel = ConsistencyLevel.SESSION;
+            }
+            if (cosmosClient != null) {
+                closeClient();
+            }
+            getLogger().debug("Creating CosmosClient");
+            createDocumentClient(uri, accessKey, clevel);
+        }
+        getCosmosDocumentContainer(context);
+    }
+
+    protected void createDocumentClient(final String uri, final String accessKey, final ConsistencyLevel clevel) throws Exception {
+        try {
+            ConnectionPolicy connectionPolicy = ConnectionPolicy.getDefaultPolicy();
+            this.cosmosClient = new CosmosClientBuilder()
+                .setEndpoint(uri)
+                .setKey(accessKey)
+                .setConnectionPolicy(connectionPolicy).setConsistencyLevel(clevel)
+                .buildClient();
+        } catch (Exception e) {
```

Comment:
Is there a more narrow Exception to use here. Also you probably don't need to specify `this.getClass().getName()` since the stack trace should capture that info.

Reply:
catching NullPointerException and removed this.getClass().getName()